codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/CargoDependency.kt

```diff
@@ -300,7 +300,7 @@ data class CargoDependency(
         val Tokio: CargoDependency =
             CargoDependency(
                 "tokio",
-                CratesIo("1.23.1"),
+                CratesIo("=1.37.0"),
```

Comment:
For future context we are seeing an unrelated [issue](https://github.com/smithy-lang/smithy-rs/actions/runs/9351952926/job/25740638098?pr=3677) with `tokio-macros:2.3.0` (released 4 days ago) which is blocking this fix:

```
 EndpointBuiltInsDecoratorTest > endpointUrlBuiltInWorksEndToEnd() FAILED
    CommandError(output=Unexpected exception thrown when executing subprocess:
    CommandError(output=Command Error
    error: failed to select a version for the requirement `tokio-macros = "~2.3.0"`
    candidate versions found which didn't match: 2.2.0, 2.1.0, 2.0.0, ...
    location searched: crates.io index
    required by package `tokio v1.38.0`
        ... which satisfies dependency `tokio = "^1.23.1"` (locked to 1.38.0) of package `test_smithy-test10366399645594046650 v1.0.0 (/home/localbuild/.local/share/smithy-test-workspace/smithy-test10366399645594046650)`
    if you are looking for the prerelease package it needs to be specified explicitly
        tokio-macros = { version = "0.2.0-alpha.6" }
    perhaps a crate was updated and forgotten to be re-vendored?

    ))
        at app//software.amazon.smithy.rust.codegen.core.util.ExecKt.runCommand(Exec.kt:49)
        at app//software.amazon.smithy.rust.codegen.core.util.ExecKt.runCommand$default(Exec.kt:15)
        at app//software.amazon.smithy.rust.codegen.core.testutil.CodegenIntegrationTestKt.codegenIntegrationTest(CodegenIntegrationTest.kt:56)
        at app//software.amazon.smithy.rust.codegen.client.testutil.ClientCodegenIntegrationTestKt.clientIntegrationTest(ClientCodegenIntegrationTest.kt:43)
        at app//software.amazon.smithy.rust.codegen.client.testutil.ClientCodegenIntegrationTestKt.clientIntegrationTest$default(ClientCodegenIntegrationTest.kt:19)
        at app//software.amazon.smithy.rustsdk.TestUtilKt.awsSdkIntegrationTest(TestUtil.kt:46)
        at app//software.amazon.smithy.rustsdk.TestUtilKt.awsSdkIntegrationTest$default(TestUtil.kt:42)
        at app//software.amazon.smithy.rustsdk.EndpointBuiltInsDecoratorTest.endpointUrlBuiltInWorksEndToEnd(EndpointBuiltInsDecoratorTest.kt:82)
```

We are going to pin to `1.37.0` which uses a previous version of `tokio-macros` to unblock releasing this fix and later revisit removing this explicit pin.